### DISCUSSION TITLE
Fire TV Stick: MXPlayer gets odd "Max streams has been reached" error starting up only 1 stream

### DISCUSSION THREAD

**Discussion by DVoorhees1957** [GerryDazoo/Slinger#259] [Q&A]
**NOTE: per my final post/reply on this thread, I'm giving up on trying to use MXPlayer. It just has too many unresolvable problems when trying to use it with Slinger**

I started testing my remote configuration, which is as follows: VLC or MXPlayer running on my Fire TV Stick 4K, talking to slingbox_server.py running on my Win7 laptop (both running on the same LAN), communicating with a Slingbox 350 on a completely different network. When I use VLC, the stream starts up very cleanly. When I use MXPlayer, I get 11 or 12 "Max streams has been reached" errors per the following log - which is weird, because I'm only trying to start up a single stream. Eventually, I get the correct video and audio stream via MXPlayer, but I'm wondering what these error messages are about?

C:\Tools\Slinger>slingbox_server.py config-R.ini
Version : 3.08d Running on Windows-7-6.1.7601-SP1 pid= 3944
Using config file config-R.ini
Connection Manager Running on port 8080 with 4 max streams using URL slingbox.
BOXES [('sb350', 'sb350')]
Streamer Running:  4 config-R.ini sb350 sb350 8080 8388608
starting up on  port 8080 8388608
Checking for slingbox at ('107.214.221.167', 5201)
('107.214.221.167', 5201) OK
sb350 Using slingbox at  ('107.214.221.167', 5201)
Streamer:  sb350 Waiting for first stream, flushing any IR requests that arrive
while not connected to slingbox
Building page for sb350
Reading Custom Remote definition from 350-remote.txt
 * Serving Flask app 'slingbox_server'
 * Debug mode: off

11/09/2022, 13:35:24.434   Streaming request from ('192.168.0.68', 40502)
Error: Can't find streamer for 8080 in dict_keys(['sb350'])
11/09/2022, 13:35:26.958   Streaming request from ('192.168.0.68', 40504)

Slinginfo  "350/500/M1/M2" 12 60 107.214.221.167 5201 3072 2
sb350 Starting Stream for  192.168.0.68:40504
Connecting... ('107.214.221.167', 5201) Control
sb350 Selecting Video Source 1
Keycodes= [1, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 22, 24,
25, 26, 27, 28, 29, 30, 31, 32, 34, 35, 37, 38, 39, 40, 41, 42, 43, 44, 46, 47,
51]
VideoParameters: Resolution= 12 FrameRate= 60 VideoBandwidth= 2000 VideoSmoothne
ss= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('107.214.221.167', 5201) Stream
sb350 Stream started at 11/09/2022, 13:35:30.407  749 0
11/09/2022, 13:35:37.2   Streaming request from ('192.168.0.68', 40506)
sb350 Got Streamer Control Message STREAM
sb350 New Stream Starting 0
11/09/2022, 13:35:40.348   Streaming request from ('192.168.0.68', 40508)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
11/09/2022, 13:35:40.619   Streaming request from ('192.168.0.68', 40510)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
.11/09/2022, 13:35:41.075   Streaming request from ('192.168.0.68', 40512)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
11/09/2022, 13:35:41.554   Streaming request from ('192.168.0.68', 40514)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
11/09/2022, 13:35:42.190   Streaming request from ('192.168.0.68', 40516)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
11/09/2022, 13:35:42.763   Streaming request from ('192.168.0.68', 40518)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
11/09/2022, 13:35:43.248   Streaming request from ('192.168.0.68', 40520)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
11/09/2022, 13:35:43.702   Streaming request from ('192.168.0.68', 40522)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
11/09/2022, 13:35:44.312   Streaming request from ('192.168.0.68', 40524)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
11/09/2022, 13:35:44.839   Streaming request from ('192.168.0.68', 40526)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
11/09/2022, 13:35:45.48   Streaming request from ('192.168.0.68', 40528)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
11/09/2022, 13:35:45.874   Streaming request from ('192.168.0.68', 40530)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
11/09/2022, 13:35:46.402   Streaming request from ('192.168.0.68', 40532)
sb350 Got Streamer Control Message STREAM
sb350 Max streams 2 for this slingbox has been reached. Not starting connection
...11/09/2022, 13:36:21.033  sb350 Stream Terminated for  192.168.0.68:40506 [Wi
nError 10035] A non-blocking socket operation could not be completed immediately

The related config.ini file contents are as follows:

[SLINGBOXES]
sb350=sb350

[sb350]
sbtype="350/500/M1/M2"
password=<350 admin pswd>
ipaddress=107.214.221.167
port=5201
name=sb350
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=2
;StartChannel=155
;RemoteLock=yes
Remote=350-remote.txt

[SERVER]
port=8080
maxremotestreams=4
enableremote=yes

What's going on that's causing these error messages, and is there a way for me to avoid getting them?

**Comment by GerryDazoo** [comment]
But from the logs it's not just 1 stream. MX player is sending 10+ separate streaming requests. (notice how the port number keeps insreasing)
I think this is an MX player issue. I've seen this behaviour before but I don't know how to reproduce it. Any way it looks like my software is working as designed. 
Is it possible MX player isn't waiting long enough before he retries.

**Reply by DVoorhees1957** [reply]
@GerryDazoo Just getting back to this... I haven't been able to figure out if there's an MX player setting that controls the wait time in between retries, so I can't really fix things that way. (Even better, I'd love to be able to disable MX player retries entirely - but haven't had any luck finding such a setting.)

So, on the Slinger side, is there a way I can simply "boot" (i.e., force close) one or more of the streams that Slinger thinks are active, but aren't really being used? That obviously implies the need to refer to open streams by either stream number, or by port...

**Reply by GerryDazoo** [reply]
No, but I think sooner or later the streams should timeout. Just to make sure we're on the same page can to upgrade to the latest server code.
 Although I'm not ruling out a server bug getting confused about the number of active streams.
But you can restart the server from the remote.

**Reply by robosmith** [reply]
@GerryDazoo I've also reported numerous streams alive on Slinger when using MxPlayer that disconnects repeatedly. 
I guess MxPlayer does not report a dropped stream to Slinger.
How do you restart the server from the remote?
Could the remote also be used to cleanup inactive streams? Or all streams when they are all inactive?

**Reply by GerryDazoo** [reply]
I think the stream will drop once you use up all available buffering assuming MXPlayer isn't doing something bad.
Click the "Restart" on Remote. 
How easy is this to reproduce?

**Reply by DVoorhees1957** [reply]
I can pretty much reproduce it at will in my current configuration (MX player running on a Fire TV Stick 4K, Slingbox Server running on a Windows 7 laptop on the same local LAN as the Fire TV Stick, with maxstreams=2, and the Slingbox running remotely). With v3.08e, the symptoms are as I originally described: 10 to 20 attempted connections from MX player, 1 of which finally works - but even though only 1 actual stream is running, Slinger reports the max (2 streams) as running. With v3.08f, the symptoms are a little bit different, but ultimately the same "max streams has been reached" message occurs.

And in either case, when I try to open what should be the actual 2nd stream (using SMPlayer on my laptop), I get a connection rejected error due to the max streams supposedly currently being used...

**Reply by regmar505** [reply]
Same behavior for me as described above with MX Player running on a Fire Stick. Slinger reports 2 streams but after a while, the second stream goes away and everything is back to normal.

**Reply by robosmith** [reply]
@GerryDazoo the remote.txt file shows a text entry: 'Restart' for the restart cmd, but the html seems to only work with a numerical value. Is there a numerical value for restart or any other way to create a restart button in the html file?

**Reply by GerryDazoo** [reply]
OK, I've got two Firesticks with MXPlayer installed and they work fine. The version number is mx V1.36.11. Is it possible that you've got some other configuration issues on your Player? Like maybe it's trying to preload up a bunch of stuff in a playlist? Maybe try uninstall/reinstall?

To answer your second question, I'm sure it's possible.  Those numbers are really just strings. Try changing one of the numbers to Restart in the html code and see what happens. But I don't support any of the third party HTML remotes. You really need to be talking to the author.

**Reply by robosmith** [reply]
@GerryDazoo MxPlayer has been working better lately. I was only getting orphaned streams when it was repeatedly disconnecting, and that was correlated mostly with prime time use; suggesting that it was caused by congestion on the Internet. The result was from repeatedly trying to reconnect to the Slinger stream and often failing on the first try. 2-3 subsequent reconnection attempts usually succeeded unless there was an abnormal number of orphaned streams still alive on Slinger.

As far as the Restart cmd, entering "Restart" in place of a numerical value ("36" for exit) caused an exception to be thrown. I will continue to experiment with that.

**Reply by DVoorhees1957** [reply]
I have the same version of MXPlayer, v1.36.11. The Fire TV is a Fire TV Stick 4K, although that shouldn't matter. Nothing else is getting preloaded; in fact, I only use MXPlayer with Slinger, not for any other purpose, When I go to Network Stream from the menu, the only thing it shows is the Slinger URL that I've used before (http://192.168.0.67:8080/slingbox/sb500).

As I said previously, I haven't been able to find anything in the MXPlayer settings to "tweak" with respect to a connection timeout and/or retry interval. I'd gladly use another app, but the only one I know about for the Fire TV is VLC Player, and that has a horrible audio lag relative to the video.

**Reply by robosmith** [reply]
@GerryDazoo I changed the Exit button from "36" to "Restart". Pressing it twice after connection log:

Starting remote control stream handler for  ('76.167.67.64', 60252) to port 12051
Remote Control Connected
Key Data ('exit', 'Restart')
Remote exit Restart
[2023-01-04 17:57:12,107] ERROR in app: Exception on /Remote/Bob [POST]
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3/dist-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/slingbox/slingbox_server.py", line 1586, in button
    int(keycode).to_bytes(1, byteorder='little') + bytes(client, 'utf-8'))
ValueError: invalid literal for int() with base 10: 'Restart'
Exiting Remote Control Stream Handler for ('76.167.67.64', 60252)
.01/04/2023, 17:57:19.121   RemoteControl connection from ('76.167.67.64', 60258)

Starting remote control stream handler for  ('76.167.67.64', 60258) to port 12051
Remote Control Connected
Key Data ('exit', 'Restart')
Remote exit Restart
[2023-01-04 17:57:19,124] ERROR in app: Exception on /Remote/Bob [POST]
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3/dist-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/slingbox/slingbox_server.py", line 1586, in button
    int(keycode).to_bytes(1, byteorder='little') + bytes(client, 'utf-8'))
ValueError: invalid literal for int() with base 10: 'Restart'
Exiting Remote Control Stream Handler for ('76.167.67.64', 60258)
01/04/2023, 17:57:22.065   RemoteControl connection from ('76.167.67.64', 60272)

Starting remote control stream handler for  ('76.167.67.64', 60272) to port 12051
Remote Control Connected
Key Data ('exit', 'Restart')
Remote exit Restart
[2023-01-04 17:57:22,068] ERROR in app: Exception on /Remote/Bob [POST]
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3/dist-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/slingbox/slingbox_server.py", line 1586, in button
    int(keycode).to_bytes(1, byteorder='little') + bytes(client, 'utf-8'))
ValueError: invalid literal for int() with base 10: 'Restart'
Exiting Remote Control Stream Handler for ('76.167.67.64', 60272)
end log

It seems the code processing the button tries to convert "Restart" to int and throws exception.

**Reply by GerryDazoo** [reply]
The code checks the key
Key Data ('exit', 'Restart') -> Key Data('Restart', 'Restart')

**Reply by robosmith** [reply]
@GerryDazoo thanks, it's working now.

**Reply by GerryDazoo** [reply]
Robo, In the next release I've defined a special keycode of -1 for restart to make the implementation more consistent.

**Comment by DVoorhees1957** [comment]
So, back to the original topic: I still have issues with MXPlayer running on a Fire TV Stick, with Slinger running on a Win7 laptop. The Fire TV Stick and the laptop are both running on the same LAN. MXPlayer was specifically installed to be the media player for Slinger, and it has no other uses on my Fire TV. I'm basically using the default installation for MXPlayer; I haven't "configured" it to do anything special, other than it has "learned" the URL to access the Slingbox (via Slinger) under the Menu -> Network Stream entry. The Fire TV Stick is using a wifi connection to communicate with the router, as is the laptop. There's not an easy way to get a wired LAN connection for the Fire TV Stick, but that's pretty easy to do with the laptop.

I can't tell whether MXPlayer is the culprit (e.g., issuing multiple connection requests when the initial request is still pending), or if something's going wrong on the Slinger side of things. I don't have any easy way to trace the interaction between MXPLayer and Slinger, although I suspect someone could guide me through using Wireshark to do a packet capture (if it comes to that - I hope that it doesn't). The fact that Gerry is using MXPlayer on a Fire TV Stick with no apparent issues leads me to believe it's something with my configuration. I'd like to sort this problem out, because MXPlayer works reasonably well other than this "multiple attempted connection" problem. It works much better than VLC Player, in particular.

**Reply by GerryDazoo** [reply]
The logs you posted to me are pretty conclusive that MXPlayer is starting multiple streams without closing the previous ones.  I have seen this behaviour once before but I am unable to reproduce it now.   Maybe there is more than one thing going wrong. ie. It times out connecting for some reason and then there's a bug that doesn't properly close the old connection. Maybe something up with your network. Do you have an Android phone laying around? You can put MXPlayer on that and see what happens to help narrow done the problem to the firestick. Maybe the FS wireless connection is not up to snuff . Can you move it closer to your router? Try both 2.4 and 5 GHz. 
We can also try pointing your FS to a different Slinger. If it is timing out maybe we can try pointing your Slinger to a difference Slingbox to rule out your hardware. So many things to try, but I would like to see you on the air.
PM me and I we can arrange a Zoom.

**Reply by DVoorhees1957** [reply]
Sadly, I don't have an Android phone to test with, only an iPhone. I do, however, have a VirtualBox instance of an Android OS (I don't remember what version). but I'm not sure that would really help, since the VirtualBox stuff might confound/compound the networking issues at play.

My laptop is probably only 45 feet from my router, and the Fire TV Stick is only 2 feet from the same router. I could move my laptop to be in the same room as my router and Fire TV Stick, if you think that would help. My laptop is currently connected via 2.4GHz, because that signal is stronger where my laptop is normally located - but again, I could change all of that. You said you have 2 Fire Sticks that are both working flawlessly using MXPlayer; are they both wifi-connected, or does one (or both) use some sort of wired LAN connection? All of this feels, instinctively, like some sort of networking problem between the Fire TV (running MXPlayer) and my laptop (running Slinger), but it's hard to believe that I would be able to connect at all if that were really the case. On the Fire TV side, the initial connection request sits and spins for anywhere between 15 seconds and 30 seconds before the connection completes - and on the laptop, there is just a boatload of separate connection requests, which tie up all of the "max streams" - even though only one of them completes successully. I guess it's the latter that's the head scratcher for me: Slinger clearly sees that there are lots of connection requests (anywhere between 10 and 30 before the "real" connection takes place), but why does it think that two connections were made, instead of just one?

Oh, and to follow up: my laptop (45 feet from my router) has no issues whatsoever running SMPlayer as the media player - so the distance between Slinger and the router isn't the issue; if distance is at all an issue (from a networking perspective), it only relates to the distance/lag between the media player (Fire TV) and Slinger (on my laptop)...

**Reply by GerryDazoo** [reply]
Three things to try,
For a test set maxstreams to one. that will cleanup the logs and help see what going on. 
You can fix the audio lag on VLC in the Firestick setting, down in the preferences, there's a place to make that adjustment. I'm not in front of my TV right now to tell you exactly where it is but it is there.
Did you try reinstalling MXPlayer?

**Reply by DVoorhees1957** [reply]
Up until now, I didn't reinstall MXPlayer simply because it hadn't been previously installed; I installed it (post-N9) specifically for the purpose of working with Slinger, since I was having problems with audio sync in VLC. I didn't do anything during or after installation to change any settings - so uninstalling and reinstalling should leave me exactly where I am now. But I just went ahead and did that, to rule it out.

Anyway, I set maxstreams to 1, and tested with MXPlayer. It took much longer than usual for the Slingbox video to show up on my Fire TV, I'm guessing 40-50 seconds. The following is the console log:
```
C:\Tools\Slinger\308e>slingbox_server.py config.ini
Version : 3.08e Running on Windows-7-6.1.7601-SP1 pid= 4480
Using config file config.ini
Connection Manager Running on port 8080 with 1 max streams using URL slingbox.
BOXES [('sb500', 'sb500')]
Streamer Running:  1 config.ini sb500 sb500 8080 8388608
starting up on  port 8080 8388608
Checking for slingbox at ('<removed_ip_address>', 55206)
('<removed_ip_address>', 55206) OK
sb500 Using slingbox at  ('<removed_ip_address>', 55206)
Streamer:  sb500 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Building page for sb500
 * Serving Flask app 'slingbox_server'
 * Debug mode: off
01/06/2023, 08:42:56.156   Streaming request from ('192.168.0.148', 34120)
Error: Can't find streamer for 8080 in dict_keys(['sb500'])
01/06/2023, 08:42:56.737   Streaming request from ('192.168.0.148', 34124)

Slinginfo  "350/500/M1/M2" 12 30 <removed_ip_address> 55206 3072 1
sb500 Starting Stream for  192.168.0.148:34124
Connecting... ('<removed_ip_address>', 55206) Control
sb500 Selecting Video Source 2
Keycodes= [1, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 22, 24, 25, 26, 27, 28, 29, 30, 31, 32, 34, 35, 37, 38, 39, 40, 41, 42, 43, 44, 46, 47, 51]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 1000 VideoSmoothness= 0 IframeRate= 5 AudioBitRate= 64
Connecting... ('<removed_ip_address>', 55206) Stream
sb500 Stream started at 01/06/2023, 08:43:00.328  744 0
01/06/2023, 08:43:08.498   Streaming request from ('192.168.0.148', 34128)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:08.553   Streaming request from ('192.168.0.148', 34130)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
.01/06/2023, 08:43:11.590   Streaming request from ('192.168.0.148', 34132)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:14.656   Streaming request from ('192.168.0.148', 34134)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:17.70   Streaming request from ('192.168.0.148', 34136)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
.01/06/2023, 08:43:20.746   Streaming request from ('192.168.0.148', 34138)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:23.769   Streaming request from ('192.168.0.148', 34140)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:26.78   Streaming request from ('192.168.0.148', 34142)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:29.819   Streaming request from ('192.168.0.148', 34144)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
.01/06/2023, 08:43:32.835   Streaming request from ('192.168.0.148', 34150)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:35.88   Streaming request from ('192.168.0.148', 34152)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:38.900   Streaming request from ('192.168.0.148', 34154)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:39.091   Streaming request from ('192.168.0.148', 34156)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:39.521   Streaming request from ('192.168.0.148', 34158)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:39.710   Streaming request from ('192.168.0.148', 34160)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:39.873   Streaming request from ('192.168.0.148', 34162)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:39.955   Streaming request from ('192.168.0.148', 34164)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:40.07   Streaming request from ('192.168.0.148', 34166)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:40.199   Streaming request from ('192.168.0.148', 34168)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:40.295   Streaming request from ('192.168.0.148', 34170)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:40.393   Streaming request from ('192.168.0.148', 34172)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:40.507   Streaming request from ('192.168.0.148', 34174)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:40.592   Streaming request from ('192.168.0.148', 34176)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:40.680   Streaming request from ('192.168.0.148', 34178)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:40.788   Streaming request from ('192.168.0.148', 34180)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
01/06/2023, 08:43:40.888   Streaming request from ('192.168.0.148', 34182)
sb500 Got Streamer Control Message STREAM
sb500 Max streams 1 for this slingbox has been reached. Not starting connection
.....01/06/2023, 08:44:29.686  sb500 Stream Terminated for  192.168.0.148:34124
[WinError 10054] An existing connection was forcibly closed by the remote host
sb500 Shutting down connections
sb500 Logging Out
Streamer:  sb500 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
```
And here is my test config.ini file:
```
[SLINGBOXES]
sb500=sb500

[sb500]
sbtype="350/500/M1/M2"
password=<removed>
ipaddress=<removed>
port=55206
name=sb500
Resolution=12
VideoBandwidth=1000
FrameRate=30
VideoSmoothness=0
VideoSource=2
maxstreams=1

[SERVER]
port=8080
maxremotestreams=1
enableremote=yes
```
I noticed that the 1st connection request, on port 34120, got a "Can't find streamer" error - not sure if that's normal, or if that even relates to this issue. The 2nd request, on port 34124, is the one that worked (I can tell because that port number is referenced when I shut MXPlayer down after the video finally showed up). The weird thing is the 45-second delay; the successful request on port 34124 came in really quickly, so why did it take the video another 45 seconds to show up? Did Slinger get tied up handling all of the extraneous connection requests and temporarily "forget" to pass the video on to MXPlayer?

**Comment by DVoorhees1957** [comment]
OK, based on some additional testing and discussion with @GerryDazoo, I'm giving up on trying to use MXPlayer. For inexplicable reasons, it sends repeated connection requests to Slinger even after the 1st request has been accepted (every 3 seconds in my case, every 7 seconds or so in Gerry's case). My main reason for trying to use MXPlayer in the first place was that VLC Player had a painful lag between the video and the audio - but there's a way to deal with that, which I will post in a new discussion (discussion #487).
